iroha/src/lib.rs

```diff
@@ -160,6 +165,17 @@ impl Iroha {
             }
         });
         let peer = Arc::clone(&self.peer);
+        let message_receiver = Arc::clone(&self.message_receiver);
```

Comment:
`message_receiver` and `message_sender` are very generic names, but are used only for p2p Messages, this could lead to confusion I think. Better to rename them.